tests/test_chiral_utils.py

```diff
@@ -138,3 +139,160 @@ def test_find_chiral_bonds():
+  1  5  1  0
+  2  6  1  0
+  2  7  1  0
+M  END"""
+
+
+def test_chiral_conflict_flip():
+    # exercise case of no conflicts or a flip conflict
+
+    mol_a = Chem.MolFromMolBlock(molblock_C, removeHs=False)
+    mol_b = Chem.MolFromMolBlock(molblock_C, removeHs=False)
+
+    conf_a = mol_a.GetConformer(0).GetPositions()
+    conf_b = mol_b.GetConformer(0).GetPositions()
+
+    chiral_set_a = ChiralRestrIdxSet.from_mol(mol_a, conf_a)
+    chiral_set_b = ChiralRestrIdxSet.from_mol(mol_b, conf_b)
+
+    assert len(chiral_set_a.restr_idxs) == 4
+    assert len(chiral_set_b.restr_idxs) == 4
+
+    identity_map = np.array([(i, i) for i in range(len(conf_a))])
+
+    swap_map = np.array(identity_map)
+    swap_map[1, 1] = 2
+    swap_map[2, 1] = 1
+
+    identity_flips = find_atom_map_chiral_conflicts(identity_map, chiral_set_a, chiral_set_b, mode="flip")
+    identity_undefineds = find_atom_map_chiral_conflicts(identity_map, chiral_set_a, chiral_set_b, mode="undefined")
+    assert len(identity_flips) == 0
+    assert len(identity_undefineds) == 0
+
+    swap_map_flips = find_atom_map_chiral_conflicts(swap_map, chiral_set_a, chiral_set_b, mode="flip")
+    swap_map_undefineds = find_atom_map_chiral_conflicts(swap_map, chiral_set_a, chiral_set_b, mode="undefined")
+    assert len(swap_map_flips) == 8  # TODO: deduplicate idxs?
+    assert len(swap_map_undefineds) == 0
+
+
+def test_chiral_conflict_undefined():
+    # exercise case where atom chirality is defined in one endstate, undefined in other
+
+    mol_a = Chem.MolFromMolBlock(molblock_C, removeHs=False)
+    mol_b = Chem.MolFromMolBlock(molblock_N, removeHs=False)
+
+    conf_a = mol_a.GetConformer(0).GetPositions()
+    conf_b = mol_b.GetConformer(0).GetPositions()
+
+    chiral_set_a = ChiralRestrIdxSet.from_mol(mol_a, conf_a)
+    chiral_set_b = ChiralRestrIdxSet.from_mol(mol_b, conf_b)
+
+    assert len(chiral_set_a.restr_idxs) == 4
+    assert len(chiral_set_b.restr_idxs) == 0
+
+    partial_map = np.array([(i, i) for i in range(4)])
+    partial_map_flips = find_atom_map_chiral_conflicts(partial_map, chiral_set_a, chiral_set_b, mode="flip")
+    partial_map_undefineds = find_atom_map_chiral_conflicts(partial_map, chiral_set_a, chiral_set_b, mode="undefined")
+    assert len(partial_map_flips) == 0
+    assert len(partial_map_undefineds) == 1
+
+
+def test_chiral_conflict_mixed():
+    # test case containing both a flip and a partial undefined
+
+    mol_a = Chem.MolFromMolBlock(molblock_CC, removeHs=False)
+    mol_b = Chem.MolFromMolBlock(molblock_CN, removeHs=False)
+
+    conf_a = mol_a.GetConformer(0).GetPositions()
+    conf_b = mol_b.GetConformer(0).GetPositions()
+
+    chiral_set_a = ChiralRestrIdxSet.from_mol(mol_a, conf_a)
+    chiral_set_b = ChiralRestrIdxSet.from_mol(mol_b, conf_b)
+
+    assert len(chiral_set_a.restr_idxs) == 8
+    assert len(chiral_set_b.restr_idxs) == 4
+
+    mixed_map = np.array([[i, i] for i in range(mol_b.GetNumAtoms())])
+
+    mixed_map[2, 0] = 3
```

Comment:
nit: # swap hydrogens part of methyl group to flip chirality

Reply:
Added in https://github.com/proteneer/timemachine/pull/908/commits/48f909e92fef38757cfc002b3ff10ca62e131a0f